src/osd/PGBackend.cc

```diff
@@ -532,26 +532,25 @@ void PGBackend::trim_rollback_object(
 
 PGBackend *PGBackend::build_pg_backend(
   const pg_pool_t &pool,
-  const OSDMapRef curmap,
+  const map<string,string>& profile,
   Listener *l,
   coll_t coll,
   ObjectStore::CollectionHandle &ch,
   ObjectStore *store,
   CephContext *cct)
 {
+  ErasureCodeProfile ec_profile = profile;
```

Comment:
tbh the error message (after i changed constness) looked like a compiler bug to me... it would build fine locally (fedora) but the make check bot failed.  I fought with this for several hours over a several days before doing this to make the error go away.